src/legacy/js/functions/_createWorkspace.js

```diff
@@ -28,9 +28,9 @@ function createWorkspace(path, collectionId, menu, collectionData, stopEventList
 
         Florence.globalVars.pagePath = safePath;
         if (Florence.globalVars.welsh !== true) {
-            document.cookie = "lang=" + "en;path=/";
+            document.cookie = "lang=" + "en;path=/" + ";domain=" + location.hostname;
         } else {
-            document.cookie = "lang=" + "cy;path=/";
+            document.cookie = "lang=" + "cy;path=/" + ";domain=" + location.hostname;
```

Comment:
Could we avoid duplication here please? If the only difference is the language value we could use a local variable.

Also there is no need to concatenate 2 strings?
```suggestion
            document.cookie = "lang=" + "cy;path=/;domain=" + location.hostname;
```